src/PostgREST/DbStructure.hs

```diff
@@ -149,14 +149,16 @@ decodeProcs =
     parseArgs = mapMaybe parseArg . filter (not . isPrefixOf "OUT" . toS) . map strip . split (==',')
 
     parseArg :: Text -> Maybe PgArg
-    parseArg a =
-      let arg = lastDef "" $ splitOn "INOUT " a
-          (body, def) = breakOn " DEFAULT " arg
+    parseArg arg =
+      let isVariadic = isPrefixOf "VARIADIC " $ toS arg
+          argNoInout = lastDef "" $ splitOn "INOUT " arg
+          argNoVariadic = lastDef "" $ splitOn "VARIADIC " argNoInout
+          (body, def) = breakOn " DEFAULT " argNoVariadic
```

Comment:
Can be improved to:
```suggestion
          -- argmode can be IN, OUT, INOUT, or VARIADIC
          argNoMode = lastDef "" $ splitOn (if isVariadic then "VARIADIC " else "INOUT ") arg
          (body, def) = breakOn " DEFAULT " argNoMode
```
(Not possible to declare a parameter like `func(INOUT VARIADIC a..`, only one argmode at a time. Check https://www.postgresql.org/docs/current/sql-createfunction.html#id-1.9.3.67.6)

`pgaVar` could take a new type but I think is fine for now.